core/templates/dev/head/pages/exploration_player/PretestEngineService.js

```diff
@@ -110,97 +141,44 @@ oppia.factory('PretestEngineService', [
-        if (!interactionId) {
-          return null;
-        }
-
-        return ExplorationHtmlFormatterService.getInteractionHtml(
-          interactionId,
-          _getCurrentStateData().interaction.customizationArgs,
-          true,
-          labelForFocusTarget);
-      },
-      getNextInteractionHtml: function(labelForFocusTarget) {
-        var interactionId = _getNextStateData().interaction.id;
-
-        return ExplorationHtmlFormatterService.getInteractionHtml(
-          interactionId,
-          _getNextStateData().interaction.customizationArgs,
-          true,
-          labelForFocusTarget);
-      },
-      getCurrentInteraction: function() {
-        return _getCurrentStateData().interaction;
-      },
-      isInteractionInline: function() {
-        var interactionId = _getCurrentStateData().interaction.id;
-        return (
-          !interactionId ||
-          INTERACTION_SPECS[interactionId].display_mode ===
-            INTERACTION_DISPLAY_MODE_INLINE);
-      },
       isNextInteractionInline: function() {
         var interactionId = _getNextStateData().interaction.id;
         return (
           !interactionId ||
           INTERACTION_SPECS[interactionId].display_mode ===
             INTERACTION_DISPLAY_MODE_INLINE);
       },
-      getCurrentInteractionInstructions: function() {
-        var interactionId = _getCurrentStateData().interaction.id;
-        return (
-          interactionId ? INTERACTION_SPECS[interactionId].instructions : '');
-      },
       getNextInteractionInstructions: function() {
         var interactionId = _getNextStateData().interaction.id;
         return (
           interactionId ? INTERACTION_SPECS[interactionId].instructions : '');
       },
-      getNextInteraction: function() {
-        return _getNextStateData().interaction;
-      },
-      isCurrentStateTerminal: function() {
-        var interactionId = _getCurrentStateData().interaction.id;
-        return (
-          interactionId && INTERACTION_SPECS[interactionId].is_terminal);
-      },
       isNextStateTerminal: function() {
         var interactionId = _getNextStateData().interaction.id;
         return (
           interactionId && INTERACTION_SPECS[interactionId].is_terminal);
       },
-      isStateShowingConceptCard: function() {
-        return false;
-      },
-      getLanguageCode: function() {
-        return pretestQuestions[currentIndex].getLanguageCode();
-      },
       getHints: function() {
         return _getCurrentStateData().interaction.hints;
       },
-      doesInteractionSupportHints: function() {
-        return (
-          !INTERACTION_SPECS[
-            _getCurrentStateData().interaction.id].is_terminal &&
-          !INTERACTION_SPECS[_getCurrentStateData().interaction.id].is_linear);
-      },
       getSolution: function() {
         return _getCurrentStateData().interaction.solution;
       },
+      getLanguageCode: function() {
```

Comment:
Same notes re: a lot of these things should be defined on, and retrieved from, the Card object.

Generally these engine services should have almost just a single function: computeNextResponse(previousData). Maybe a couple more things along the same lines -- but not much more than that.

Reply:
Yeah, most of the 'current/next' state functions are taken from card objects. The main problem is that hints and solutions button directive and service also use some functions and they don't have a copy of the card object. For the directive, I can pass one as attribute, but what about service? Shall I jjst import transcript service and take it?

Reply:
Done (though language code is still there itself for now).